### ISSUE TITLE
Either making "ghc-8.0" the new master, or making it the GitHub default branch

### ISSUE THREAD

**Issue by mgsloan** [ghcjs/ghcjs#526]
Since `ghc-8.0` is the current bleeding edge of development, I think it makes sense to have it bet the GitHub default branch.  This means that:

* It will be the branch visible when you visit the repo

* It will be the default destination for PRs

I think this will be clearer to contributors!

**Comment by spl** [comment]
@mgsloan: Thanks for starting the discussion.

I think what you did for the transition to `improved-base` was useful. Granted there were more simultaneous changes among multiple packages, so it won't be so much, but documenting the branches in various places such as the READMEs was helpful.

I agree that, if the `ghc-8.0` branch is ready to be the focus of PRs, the switch should be made.

Is it really necessary to force-push to `master`? Can `ghc-8.0` be rebased? Have you tried looking at the merge conflicts to see they can be resolved?

If you need help or feedback, I'm happy to give both.

**Comment by mgsloan** [comment]
Cool, sounds good @spl !

I've talked to luite about it, and `master` and `ghc-8.0` are diverged and are unlikely to converge.  So I think `ghc-8.0` should be the new master.

Thoughs, @luite ?  Should we just flip the switch and turn `ghc-8.0` into master?

**Comment by luite** [comment]
Yeah probably a good idea, make `ghc-7.10` branches first for everything relevant.

I'll update my snapshot builder once this is done.

GHCJSi support is still coming, but making the `ghc-8.0` branch wait for that would probably do more harm than good at the moment...

**Comment by Ericson2314** [comment]
I've mentioned this a few times in the past, but we could have a `stable` that (for now) aliases `ghc-7.0`, and would be merged into `ghc-8.0` when that is ready for release. This allows master to keep its role for active development perpetually so we don't run into this issue again.

**Comment by spl** [comment]
@luite: What's the next step? I see you have [ongoing work](https://github.com/spl/ghcjs/compare/merged-ghc-8.0-to-master...ghcjs:ghc-8.0) on the `ghc-8.0` branch. Do you want to make the `ghc-7.10` branch and then either merge #528 or do @mgsloan's swap?

**Comment by mgsloan** [comment]
Yes, I think it'd be really good to get the branches straightened out.  Then we'll also want to merge or close all existing PRs.

I would just do the rename, but I don't want to screw up your development flow, @luite!  So it'd be best if you could make master the "real development" tip, and we can let the `ghc-7.10` branch be a historical thing.

Currently, there are two branches that are never going to be merged.  It's hard to do a collaborative project with two different tips!
